src/lib/event_config.c

```diff
@@ -556,3 +556,43 @@ bool check_problem_rating_usability(const event_config_t *cfg,
 
     return result;
 }
+
+GList *expand_event_wildcard(const gchar *event_name, gsize event_len)
+{
+    GList *list = NULL;
+
+    if (event_name[event_len - 1] == '*')
+    {
+        log_info("expanding wildcard in event name '%s'", event_name);
+
+        /* List all available events. */
+        GList *event_files = get_file_list(EVENTS_DIR, "xml");
+        if (!event_files)
+            log_warning("could not list available events or none found");
+        else
+        {
+            /* For each listed event, compare its prefix with the wildcard
+             * prefix and add it to the resulting list if it matches. */
+            while (event_files)
+            {
+                file_obj_t *file = (file_obj_t *)event_files->data;
+                const gchar *file_name = fo_get_filename(file);
+
+                /* Check if the event's name matches the required prefix. */
+                if (strncmp(file_name, event_name, event_len - 1) == 0)
+                {
+                    log_debug("found matching event '%s'", file_name);
+                    list = g_list_append(list, xstrdup(file_name));
+                }
+
+                free_file_obj(file);
+                event_files = g_list_delete_link(event_files, event_files);
+            }
+        }
+    }
+    else
+        list = g_list_append(list, xstrdup(event_name));
```

Comment:
Might be better to invert the condition at the beginning and do something like `return g_list_prepend (NULL, xstrdup (event_name));`